crates/oxc_transformer/src/typescript/namespace.rs

```diff
@@ -200,107 +197,85 @@ impl<'a> TypeScript<'a> {
                         new_stmts.push(transformed);
                     }
                 }
-                Statement::ClassDeclaration(decl) => {
+                Statement::ClassDeclaration(ref decl) => {
                     is_empty = false;
                     decl.bound_names(&mut |id| {
                         names.insert(id.name.clone());
                     });
-                    new_stmts.push(Statement::ClassDeclaration(decl));
+                    new_stmts.push(stmt);
                 }
-                Statement::FunctionDeclaration(decl) => {
+                Statement::FunctionDeclaration(ref decl) => {
                     is_empty = false;
                     decl.bound_names(&mut |id| {
                         names.insert(id.name.clone());
                     });
-                    new_stmts.push(Statement::FunctionDeclaration(decl));
+                    new_stmts.push(stmt);
                 }
-                Statement::TSEnumDeclaration(enum_decl) => {
+                Statement::TSEnumDeclaration(ref enum_decl) => {
                     is_empty = false;
                     names.insert(enum_decl.id.name.clone());
-                    new_stmts.push(Statement::TSEnumDeclaration(enum_decl));
+                    new_stmts.push(stmt);
                 }
-                Statement::VariableDeclaration(decl) => {
+                Statement::VariableDeclaration(ref decl) => {
                     is_empty = false;
                     decl.bound_names(&mut |id| {
                         names.insert(id.name.clone());
                     });
-                    new_stmts.push(Statement::VariableDeclaration(decl));
+                    new_stmts.push(stmt);
+                }
+                Statement::ExportNamedDeclaration(ref export_decl)
```

Comment:
@Dunqing This PR doesn't change the logic, but this doesn't look right to me. We can end up with `export {x}` in the output. [Playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAICsgICAgICAgIC3mMl9fWP2DOEtS29bAAVF3HDuYFci2KI8dKJg9t1AO3n4jQYRoCBjAGTVEvqeRiV%2FeJWQgA%3D%3D)

I'm not sure what the correct behaviour is, though.